index.js

```diff
@@ -14,54 +16,111 @@ module.exports.pitch = function(remainingRequest) {
 			regExp: query.regExp
 		};
 		var chunkName = loaderUtils.interpolateName(this, query.name, options);
-		var chunkNameParam = ", " + JSON.stringify(chunkName);		
+		var chunkNameParam = ", " + JSON.stringify(chunkName);
 	} else {
 		var chunkNameParam = '';
 	}
 	var result;
 	if(query.lazy) {
 		result = [
-			"module.exports = function(cb) {\n",
-			"	require.ensure([], function(require) {\n",
-			"		cb(require(", loaderUtils.stringifyRequest(this, "!!" + remainingRequest), "));\n",
+			"module.exports = function(successCallback, errorCallback) {\n",
+			"	require.ensure([], function() {\n",
+			"		successCallback(require(", loaderUtils.stringifyRequest(this, "!!" + remainingRequest), "));\n",
+			"	}, function() {\n",
+			"		if (errorCallback) errorCallback.apply(this, arguments);\n",
 			"	}" + chunkNameParam + ");\n",
-			"}"];
+			"};"];
 	} else {
 		result = [
-			"var cbs = [], \n",
-			"	data;\n",
-			"module.exports = function(cb) {\n",
-			"	if(cbs) cbs.push(cb);\n",
-			"	else cb(data);\n",
-			"}\n",
-			"require.ensure([], function(require) {\n",
-			"	data = require(", loaderUtils.stringifyRequest(this, "!!" + remainingRequest), ");\n",
-			"	var callbacks = cbs;\n",
-			"	cbs = null;\n",
-			"	for(var i = 0, l = callbacks.length; i < l; i++) {\n",
-			"		callbacks[i](data);\n",
+			"var cbs,\n",
```

Comment:
Please remove all ending `\n` in the code snippets and instead `result.join('\n')` them on line 73

Reply:
Good catch! I've made the change.